From: Hannes Reinecke <hare@suse.de>
Date: Tue, 23 Sep 2014 15:46:15 +0200
Subject: iscsi_ibft: Always display netmask
References: bnc#897960
Patch-Mainline: not yet

Some older user-space code might rely on the netmask attribute
being present, so we should always display it.
This fixes a regression introduced by commit
0b2eb3c4060a16f3ec11a4d6d4c934e7e5d5334f.

Signed-off-by: Hannes Reinecke <hare@suse.de>
---
 drivers/firmware/iscsi_ibft.c | 8 +-------
 1 file changed, 1 insertion(+), 7 deletions(-)

diff --git a/drivers/firmware/iscsi_ibft.c b/drivers/firmware/iscsi_ibft.c
index 611d47c..0a4073d 100644
--- a/drivers/firmware/iscsi_ibft.c
+++ b/drivers/firmware/iscsi_ibft.c
@@ -452,14 +452,8 @@ static umode_t ibft_check_nic_for(void *data, int type)
 			rc = S_IRUGO;
 		break;
 	case ISCSI_BOOT_ETH_PREFIX_LEN:
-		if (nic->subnet_mask_prefix)
-			rc = S_IRUGO;
-		break;
 	case ISCSI_BOOT_ETH_SUBNET_MASK:
-		if (!memcmp(nic->ip_addr, nulls, 10) &&
-		    (nic->ip_addr[10] == 0xff) &&
-		    (nic->ip_addr[11] == 0xff) &&
-		    nic->subnet_mask_prefix)
+		if (nic->subnet_mask_prefix)
 			rc = S_IRUGO;
 		break;
 	case ISCSI_BOOT_ETH_ORIGIN:
-- 
1.8.5.2

